### PR TITLE
Move files out of RunTestOnHost and remove RunTestOnHost and RunTestOnVm.

### PR DESCRIPTION
**Pull request by JimmyCYJ**
I will rename the scripts to make them consistent in another PR.

### CODE REVIEW

File: vagrant-kubernetes-istio/README.md [hunk 1]
```diff
@@ -1,19 +1,103 @@
-# vagrant-kubernetes-istio
+The following instruction details the steps required to run E2E tests with a Vagrant VM on your local machine, so you can test and debug locally.
 
-Set up Kubernetes on VM with vagrant for Istio e2e tests, so that we can run Istio e2e tests on local machine [issue](https://github.com/istio/istio/issues/4536)
+# Benefits:
+1. Set up a local vagrant VM Environment once and run "make e2e_all" to run E2E tests from your development environment.
+1. No need to worry about kubernetes cluster setup. The scripts take care of that.
+1. TODO: Debug tests right from your development environment.
```
Comment (gargnupur) [inline vagrant-kubernetes-istio/README.md]: Can remove this now?

File: vagrant-kubernetes-istio/README.md [hunk 2]
```diff
@@ -1,19 +1,103 @@
-# vagrant-kubernetes-istio
+The following instruction details the steps required to run E2E tests with a Vagrant VM on your local machine, so you can test and debug locally.
 
-Set up Kubernetes on VM with vagrant for Istio e2e tests, so that we can run Istio e2e tests on local machine [issue](https://github.com/istio/istio/issues/4536)
+# Benefits:
+1. Set up a local vagrant VM Environment once and run "make e2e_all" to run E2E tests from your development environment.
```
Comment (gargnupur) [inline vagrant-kubernetes-istio/README.md]: should we add that local_cluster flag too here?

File: vagrant-kubernetes-istio/README.md [hunk 3]
```diff
@@ -1,19 +1,103 @@
...
+## 4. Run tests!
+### 1. Run tests on host machine
+You can issue test command on your host machine.
+E.g.
+```bash
+cd $ISTIO/istio
+make e2e_simple E2E_ARGS="--use_local_cluster" HUB=10.10.0.2:5000 TAG=latest
+```
+You can keep repeating this step if you made any local changes and want to run e2e tests again.
+Add E2E_ARGS="--use_local_cluster" to all your e2e tests as tests are we are running a local cluster.
```
Comment (gargnupur) [inline vagrant-kubernetes-istio/README.md]: nit: as we are running tests on a local cluster?

Comment (JimmyCYJ) [reply vagrant-kubernetes-istio/README.md]: Done.

File: vagrant-kubernetes-istio/README.md (hunk 2 above)
Comment (JimmyCYJ) [reply vagrant-kubernetes-istio/README.md]: Added.

File: vagrant-kubernetes-istio/README.md (hunk 1 above)
Comment (JimmyCYJ) [reply vagrant-kubernetes-istio/README.md]: Removed.

**Pull request merged**